x-pack/plugins/security_solution/public/management/pages/host_isolation_exceptions/view/components/delete_modal.tsx

```diff
@@ -17,125 +17,116 @@ import {
   EuiText,
 } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n/react';
-import { useDispatch } from 'react-redux';
-import { Dispatch } from 'redux';
 import { i18n } from '@kbn/i18n';
-import { useToasts } from '../../../../../common/lib/kibana';
-import { useHostIsolationExceptionsSelector } from '../hooks';
-import {
-  getDeleteError,
-  getItemToDelete,
-  isDeletionInProgress,
-  wasDeletionSuccessful,
-} from '../../store/selector';
-import { HostIsolationExceptionsPageAction } from '../../store/action';
-
-export const HostIsolationExceptionDeleteModal = memo<{}>(() => {
-  const dispatch = useDispatch<Dispatch<HostIsolationExceptionsPageAction>>();
-  const toasts = useToasts();
-
-  const isDeleting = useHostIsolationExceptionsSelector(isDeletionInProgress);
-  const exception = useHostIsolationExceptionsSelector(getItemToDelete);
-  const wasDeleted = useHostIsolationExceptionsSelector(wasDeletionSuccessful);
-  const deleteError = useHostIsolationExceptionsSelector(getDeleteError);
-
-  const onCancel = useCallback(() => {
-    dispatch({ type: 'hostIsolationExceptionsMarkToDelete', payload: undefined });
-  }, [dispatch]);
-
-  const onConfirm = useCallback(() => {
-    dispatch({ type: 'hostIsolationExceptionsSubmitDelete' });
-  }, [dispatch]);
+import { ExceptionListItemSchema } from '@kbn/securitysolution-io-ts-list-types';
+import { useMutation } from 'react-query';
+import { useHttp, useToasts } from '../../../../../common/lib/kibana';
+import { deleteOneHostIsolationExceptionItem } from '../../service';
 
-  // Show toast for success
-  useEffect(() => {
-    if (wasDeleted) {
-      toasts.addSuccess(
-        i18n.translate(
-          'xpack.securitySolution.hostIsolationExceptions.deletionDialog.deleteSuccess',
-          {
-            defaultMessage: '"{name}" has been removed from the Host isolation exceptions list.',
-            values: { name: exception?.name },
-          }
-        )
-      );
+export const HostIsolationExceptionDeleteModal = memo(
+  ({
+    item,
+    onCancel,
+  }: {
+    item: ExceptionListItemSchema;
+    onCancel: (forceRefresh?: boolean) => void;
+  }) => {
+    const toasts = useToasts();
+    const http = useHttp();
 
-      dispatch({ type: 'hostIsolationExceptionsMarkToDelete', payload: undefined });
-    }
-  }, [dispatch, exception?.name, toasts, wasDeleted]);
+    const mutation = useMutation(
+      () => {
```

Comment:
should the input to this hook (like the callback and the second argument object) be memoized with `useMemo` and/or `useMemo`?

Reply:
No, useCallback would not make any difference here, if any you would use more memory because you construct the function that you pass to this every run anyway.